src/sentry/release_health/metrics.py

```diff
@@ -606,3 +604,346 @@ def extract_row_info(row: Mapping[str, Union[OrganizationId, str]]) -> ReleaseNa
+        rv_users = self._get_users_and_crashed_users_for_overview(where, org_id)
+
+        # XXX: In order to be able to dual-read and compare results from both
+        # old and new backend, this should really go back through the
+        # release_health service instead of directly calling `self`. For now
+        # that makes the entire backend too hard to test though.
+        release_adoption = self.get_release_adoption(project_releases, environments)
+
+        rv: Dict[ProjectRelease, ReleaseHealthOverview] = {}
+
+        fetch_has_health_data_releases = set()
+
+        for project_id, release in project_releases:
+            adoption_info: ReleaseAdoption = release_adoption.get((project_id, release)) or {
+                "adoption": None,
+                "sessions_adoption": None,
+                "users_24h": None,
+                "project_users_24h": None,
+                "sessions_24h": None,
+                "project_sessions_24h": None,
+            }
+
+            total_sessions = rv_sessions.get((project_id, release, "init"))
+
+            total_users = rv_users.get((project_id, release, "init"))
+            has_health_data = bool(total_sessions)
+
+            # has_health_data is supposed to be irrespective of the currently
+            # selected rollup window. Therefore we need to run another query
+            # over 90d just to see if health data is available to compute
+            # has_health_data correctly.
+            if not has_health_data and summary_stats_period != "90d":
+                fetch_has_health_data_releases.add((project_id, release))
+
+            sessions_crashed = rv_sessions.get((project_id, release, "crashed"), 0)
+
+            users_crashed = rv_users.get((project_id, release, "crashed"), 0)
+
+            rv_row = rv[project_id, release] = {
+                "adoption": adoption_info.get("adoption"),
+                "sessions_adoption": adoption_info.get("sessions_adoption"),
+                "total_users_24h": adoption_info.get("users_24h"),
+                "total_project_users_24h": adoption_info.get("project_users_24h"),
+                "total_sessions_24h": adoption_info.get("sessions_24h"),
+                "total_project_sessions_24h": adoption_info.get("project_sessions_24h"),
+                "total_sessions": total_sessions,
+                "total_users": total_users,
+                "has_health_data": has_health_data,
+                "sessions_crashed": sessions_crashed,
+                "crash_free_users": (
+                    100 - users_crashed / total_users * 100 if total_users else None
+                ),
+                "crash_free_sessions": (
+                    100 - sessions_crashed / float(total_sessions) * 100 if total_sessions else None
+                ),
+                "sessions_errored": max(
+                    0,
+                    rv_errored_sessions.get((project_id, release), 0)
+                    - sessions_crashed
```

Comment:
This subtraction logic should get deprecated with metrics, no?
We were only doing this because with materialized views, we were getting some negatives

Reply:
I don't have the context on this, can you elaborate? I also couldn't tell you whether you'd not get negatives here.

Reply:
So basically we found that sometimes especially in the one hour materialized tables that this subtraction could result in a negative. 
I might be wrong here but I was under the impression that with metrics that wouldn't happen because you are basically counting actual numbers?

Reply:
we can start emitting an additional metric for this, but the metrics we currently extract are the bare minimum required to support the current data model.

I would note this down somewhere as a bug (perhaps in INGEST backlog), but I think it would be easier to compare backend equivalence in an automated fashion if we're returning the same exact results (even if the results are wrong). If we want to fix a bug while we compare we would need to either:

a. find a way to filter the comparison results for this (not sure, maybe there's a way)
b. fix the bug in both backends (but I think you're saying we can't fix this in sessions backend)